scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/codecs/package.scala

```diff
@@ -0,0 +1,102 @@
+package scala.meta.internal.semantic
+
+import org.scalameta.unreachable
+import scala.meta.io._
+import scala.meta.{semantic => m}
+import scala.meta.internal.semantic.{proto => p}
+
+package object codecs {
+  implicit class XtensionProtoSerializable[A](val a: A) extends AnyVal {
+    def toMeta[B](implicit ev: ProtoDecoder[B, A]): B = ev.fromProto(a)
+    def toProto[B](implicit ev: ProtoEncoder[A, B]): B = ev.toProto(a)
+    def toProtoOpt[B](implicit ev: ProtoEncoder[A, B]): Option[B] = Option(toProto[B])
+  }
+
+  implicit val AttributedSourceCodec: ProtoCodec[m.AttributedSource, p.AttributedSource] =
+    new ProtoCodec[m.AttributedSource, p.AttributedSource] {
+      override def toProto(e: m.AttributedSource): p.AttributedSource = e match {
+        case m.AttributedSource(path, names, messages, denotations, sugars) =>
+          implicit val resolvedNameEncoder = new ProtoEncoder[(m.Anchor, m.Symbol), p.ResolvedName] {
```

Comment:
I don't think we can, because they require `path` from `m.AttributedSource` to be in scope. I learned this the hard way by implementing them at the top level :)